lib/AST/RequirementMachine/RequirementMachineRequests.cpp

```diff
@@ -776,7 +776,8 @@ InferredGenericSignatureRequest::evaluate(
         WhereClauseOwner whereClause,
         SmallVector<Requirement, 2> addedRequirements,
         SmallVector<TypeBase *, 2> inferenceSources,
-        SourceLoc loc, bool isExtension, bool allowInverses) const {
+        SourceLoc loc, bool isExtension, bool allowInverses,
+        std::optional<InvertibleProtocolKind> conditionalInvertibleProtocolExtension) const {
```

Comment:
Can we consolidate this proliferation of bits into an ExtensionKind or similar?

Reply:
I could possibly consolidate `isExtension` and `conditionalInvertibleProtocolExtension`. The other place we appear to use `allowInverses` is for printing certain generic signatures in SIL for unrelated reasons, so I didn't want to disturb its existing behavior.